src/compile/data/assemble.ts

```diff
@@ -268,5 +268,38 @@ export function assembleRootData(dataComponent: DataComponent, datasets: Dict<In
     }
   }
 
+  // make the aggregate type transform in the first order
+  // to make it effective when there are more than one transforms in the data
+  for(const d of data){
+    swapTransformTypeIndex(d,"aggregate",0);
+  }
+
   return data;
 }
+
+// swap the index of transform for the type to the targetIndex
+function swapTransformTypeIndex(data:VgData,typeName:string,targetIndex:number) {
+  // if the data.transform is exist and its length >=2
```

Comment:
remove all comments that describe the code and only keep the ones that describe the why